### DISCUSSION TITLE
Component boxROI and the gravity

### DISCUSSION THREAD

**Discussion by andersonnardin** [SofaDefrost/SoftRobots#196] [Help - Building / Configuring] [answered]
Hello all,

I have a basic question regarding the component BoxROI when it is used to prevent the falling of the nodes on a scene, for example, on scenes of the tutorials. 
If it is defined such that no point of the objects is inside, the bodies were supposed to fall due to the gravity we impose, right?

I am questioning because it is not behaving this way. If you test on the tutorial pneunet for example, and define a boxROI away from bodies on the scene. For example:

`finger.addObject('BoxROI', name='boxROI', box=[-10, 40, -20, 0, 50, 20], doUpdate=False, drawBoxes=True)`

The bodies don't fall. I verified the same behavior on other more complex scenes and sometimes the position and presence of the boxROI influences the expansion of the cavity (sometimes crashing the software).

Could you confirm this behavior and let me know your thoughts about it?
Thanks!

**Comment by alxbilger** [comment, marked as answer]
Hi,

I investigated the scenes in [this folder](https://github.com/SofaDefrost/SoftRobots/tree/3ca12ebd582b1fc26c6184c3763da6caf0e137fc/docs/sofapython3/tutorials/PneunetGripper/details).

The behavior you observe when the box is away from the object is due to the `RestShapeSpringsForceField`. It uses the indices from the `BoxROI`, but the list of indices is empty. When the list of indices in `RestShapeSpringsForceField` is empty, the whole object is used. So, the springs acts on every point of the mesh. And it is much stronger than the mass. That's why the object does not move.
Remove the `RestShapeSpringsForceField` or reduce its stiffness, and you will see the object falling.

When you have such a situation, I recommend to enable `printLog` on some key components. It may help you understand what is going on.

Alex

**Comment by andersonnardin** [comment]
Hi @alxbilger 
Thank you for replying.
My question extends to the case where a cavity mesh is deformed (toward the outer of the body) when pressure is applied.


I verified the `printLog `from the `RestShapeSpringsForceField` and it is ok. 
If it is not related to the boxROI, do you have any idea why such behavior happens?

**Reply by alxbilger** [reply]
I don't understand what is happening in your scene. I would need more details. What is the role of the BoxROI and the RestShapeSpringsForceField in the cavity? What happens with and without the BoxROI?

BoxROI is just a component that select points inside a box. It should not change the behavior, unless another component change its behavior depending on the content of the box.

**Comment by andersonnardin** [comment]
the BoxROI and the RestShapeSpringsForceField don't touch the cavity space.
They are there simply to prevent the body from falling.

**Reply by alxbilger** [reply]
I you add forces on some points of your mesh, to force them to stay where they are (using RestShapeSpringsForceField), it makes sense that it influence the result of your simulation. The cavity and the springs are both applying on your object. If you don't add the springs, you will not get the same result.
